pkg/api/types.go

```diff
@@ -107,6 +109,26 @@ type AgentPoolProfile struct {
 	Role         AgentPoolProfileRole `json:"role,omitempty"`
 }
 
+type AuthProfile struct {
+	IdentityProviders []IdentityProvider `json:"identityProviders,omitempty"`
+}
+
+// IdentityProvider is heavily cut down equivalent to IdentityProvider in the
+type IdentityProvider struct {
+	Name     string         `json:"name"`
+	Provider runtime.Object `json:"provider,omityempty"`
```

Comment:
I'm very strongly tempted to use `interface{}` instead of `runtime.Object`, and `Kind string ...` instead of `metav1.TypeMeta` to avoid pulling in more k8s libraries in our API definition.  WDYT?

Reply:
In fact, I think it's essential to remove the k8s deps for the v1 definition, so please do so for api as well.